test/server/admin/clusters_handler_test.cc

```diff
@@ -243,5 +245,70 @@ fake_cluster::1.2.3.4:80::local_origin_success_rate::93.2
   EXPECT_EQ(expected_text, response2.toString());
 }
 
+TEST_P(AdminInstanceTest, TestSetHealthFlag) {
+  std::shared_ptr<Upstream::MockClusterInfo> cluster{new NiceMock<Upstream::MockClusterInfo>()};
+  Event::MockDispatcher dispatcher;
+  auto host = Upstream::makeTestHost(cluster, "tcp://127.0.0.1:9000", dispatcher.timeSource());
+  envoy::admin::v3::HostHealthStatus health_status;
+
+  // FAILED_ACTIVE_HC
+  host->healthFlagSet(Upstream::Host::HealthFlag::FAILED_ACTIVE_HC);
+  setHealthFlag(Upstream::Host::HealthFlag::FAILED_ACTIVE_HC, *host, health_status);
+  EXPECT_TRUE(health_status.failed_active_health_check());
+
+  // FAILED_OUTLIER_CHECK
+  host->healthFlagSet(Upstream::Host::HealthFlag::FAILED_OUTLIER_CHECK);
+  setHealthFlag(Upstream::Host::HealthFlag::FAILED_OUTLIER_CHECK, *host, health_status);
+  EXPECT_TRUE(health_status.failed_outlier_check());
+
+  // FAILED_EDS_HEALTH
+  host->healthFlagSet(Upstream::Host::HealthFlag::FAILED_EDS_HEALTH);
+  setHealthFlag(Upstream::Host::HealthFlag::FAILED_EDS_HEALTH, *host, health_status);
+  EXPECT_EQ(health_status.eds_health_status(), envoy::config::core::v3::UNHEALTHY);
```

Comment:
thanks for filling in all these missing tests. Was this all previously uncovered by tests? Maybe we should increase the coverage expectation.

Reply:
I don't see the specific test cover this utility, but just [one json unit test](https://github.com/envoyproxy/envoy/pull/32801/files#diff-7fc8d177a372e19dab401678b8126faea43dff984156aad5e39ca74a795f9bebR18). It is using `oncall.return(true)`, so the coverage is same (96.1).

Reply:
taking a second look.

Reply:
you are saying all the lines covered by this new test were already covered?

Reply:
For this specific bug, the coverage system does not distinguish the

```
ON_CALL(*host, healthFlagGet(Upstream::Host::HealthFlag::EDS_STATUS_DRAINING))
      .WillByDefault(Return(true));
```
or `Return(false)` by marking it was tested. If we return false, the tests should fail, since the buggy code always assume the flag is set without checking it is false or not.

Reply:
I see; we were hitting the lines, just not in a meaningful way; SGTM.